source/Dig/base.framework.entity.base.bmx

```diff
@@ -78,30 +100,26 @@ Type TEntityCollection
 
 
 	Method Add:int(obj:TEntityBase)
-		'insert does not return true or false, so we invalidate in all cases
-		rem
-		if entries.Insert(obj.GetGUID(), obj)
-			'invalidate count
-			entriesCount = -1
-
-			return TRUE
-		endif
-
-		return False
-		endrem
-
+		if not entriesID Then CreateEntriesID()
+		entriesID.Insert(obj.GetID(), obj)
 		entries.Insert(obj.GetGUID(), obj)
 		entriesCount = -1
 		return True
 	End Method
 
 
 	Method Remove:int(obj:TEntityBase)
-		if obj.GetGuid() and entries.Remove(obj.GetGUID())
+		Local objID:Int = obj.GetID()
```

Comment:
Ich glaube, der Code funktioniert nur "halb". Die Collections mit doppelten IDs sollten nicht davon ausgehen, dass bei reingegebenen Objekten die ID und GUID immer konsistent zusammenpassen.

Bsp.: o1 mit ID 1 und GUID "TADA", o2 mit ID 2 und GUID "TADA"; o1 wurde mit add hinzugefügt, Remove wird mit o2 aufgerufen.

Der if-Zweig greift nicht, weil ID 2 unbekannt ist, der ElseIf-Zweig greift, aber aus entriesID wird 2 entfernt, o1 bleibt also in einer der Maps.

Stattdessen müsste man sich erst aus der entries-Map das Objekt mir der GUID geben lassen und DESSEN id aus entriesID entfernen. Ansonsten laufen die beiden Maps auseinander.

Das könnte andere Stellen, die dasselbe Pattern verwenden auch betreffen. (Vielleicht ist das ja auch eine der seg-fault-Ursachen)

Reply:
wenn eine GUID 2x vergeben wurde ... haben wir andere Probleme.

Reply:
Naja, das passiert ganz automatisch, z.B. beim Datenbankladen (Datenbanknochmalladen).

Reply:
Der Code könnte zumindest so gehärtet werden, dass beim Hinzufügen/Löschen sichergestellt wird, dass die beiden Listen konsistent bleiben. Dann würde man zumindest sofort merken, dass man ein Problem hat.

Reply:
> Naja, das passiert ganz automatisch, z.B. beim Datenbankladen (Datenbanknochmalladen).

Kannst du mir das nochmal umschreiben, ich koch jetzt erstmal Mittag und koennte da so schneller "danach" hier einsteigen.